pingrid.py

```diff
@@ -27,6 +27,25 @@
 RAIN_PNE_COLORMAP = "[0xa020f0 [0x0000ff 35] [0x40e0d0 36] [0x32cd32 36] 0xffe465 [0xffe465 35] 0xffff00 [0xffa500 35] [0xa52a2a 36] [0x000000 36]]"
 CORRELATION_COLORMAP = "[0x000000 0x000080 [0x0000ff 25] [0x00bfff 26] [0x7fffd4 39] [0x98fb98 26] 0xffe465 [0xffe465 25] 0xffff00 [0xff8c00 38] [0xff0000 39] [0x800000 39] 0xa52a2a]"
 
+
+def sel_snap(spatial_array, lat, lng, dim_y="Y", dim_x="X"):
+    """"Selects the spatial_array's closest spatial grid center to the lng/lat coordinate.
+    Raises an exception if lng/lat is outside spatial_array domain.
+    Assumes regularly spaced dimensions.
+    Assumes dimensions ordered from low to high.
+    """
+    the_method = None
+    half_res_y = (spatial_array[dim_y][1] - spatial_array[dim_y][0]) / 2
+    half_res_x = (spatial_array[dim_x][1] - spatial_array[dim_x][0]) / 2
+    min_y = spatial_array[dim_y][0] - half_res_y
+    max_y = spatial_array[dim_y][-1] + half_res_y
+    min_x = spatial_array[dim_x][0] - half_res_x
+    max_x = spatial_array[dim_x][-1] + half_res_x
+    if lat >= min_y and lat <= max_y and lng <= min_x and lng >= max_x:
+        the_method = "nearest"
```

Comment:
This is the kind of place where you could use the `shapely` approach for bounds checking that I suggested elsewhere, to handle longitude wrapping. But I suppose there's not much point doing that in this case, since the following line is a `.sel`, whcih doesn't handle longitude wrapping.